### DISCUSSION TITLE
setting host_keys with env doesn't work as expected

### DISCUSSION THREAD

**Discussion by ziporah** [drakkan/sftpgo#1488] [Q&A] [answered]
We are running sftpgo on openshift / kubernetes
When setting the host keys pointing to files by setting the environment variables I would expect the service to load the files.
```
"SFTPGO_SFTPD__HOST_KEYS__0"                  = "/etc/sftpgo/keys/id_rsa", 
"SFTPGO_SFTPD__HOST_KEYS__1"                  = "/etc/sftpgo/keys/id_ecdsa",
"SFTPGO_SFTPD__HOST_KEYS__2"                  = "/etc/sftpgo/keys/id_ed25519",
```
the documentations states:
You can also override all the available configuration options using environment variables. SFTPGo will check for environment variables with a name matching the key uppercased and prefixed with the SFTPGO_. You need to use __ to traverse a struct.

So setting SFTPGO_SFTPD__ would get me in the sftpd configuration struct, using HOST_KEYS pushes down to the host keys and __0,__1,__2 should set the keys I would expect from reading the docs. The keys don't get read by the process however.

The correct method to load the keys is by setting a comma seperated list of values like this.
```
"SFTPGO_SFTPD__HOST_KEYS"                    = "/etc/sftpgo/keys/id_rsa,/etc/sftpgo/keys/id_ecdsa,/etc/sftpgo/keys/id_ed25519",
```
I believe this should be cleared up in the documentation.

**Comment by drakkan** [comment, marked as answer]
Hi,

host keys is a list of string not a struct and don't contain a struct inside. In the docs there is also an example to set env vars for string lists (`SFTPGO_COMMON__ACTIONS__EXECUTE_ON=upload,download`).

Setting env vars for lists as comma separated values is the standard viper behaviour and so it was is expected. Feel free to send a PR to clarify things but make sure to not confuse lists and structs. Thank you
